packages/migrations/src/migrations/migrations.test.ts

```diff
@@ -45,14 +49,88 @@ describe('migrations', () => {
           timestamp: '20220101010102',
         },
       ];
-      assert.deepEqual(getMigrationsToExecute(migrationFiles, executedMigrations), [
+      assert.deepEqual(
+        getMigrationsToExecute(migrationFiles, { excludeMigrations: executedMigrations }),
+        [
+          {
+            directory: 'migrations',
+            timestamp: '20220101010103',
+            filename: '20220101010103_testing_3.sql',
+          },
+        ],
+      );
+    });
+  });
+
+  it('handles case where beforeTimestamp is specified', () => {
+    const migrationFiles = [
+      {
+        directory: 'migrations',
+        filename: '20220101010101_testing_1.sql',
+        timestamp: '20220101010101',
+      },
+      {
+        directory: 'migrations',
+        filename: '20220101010102_testing_2.sql',
+        timestamp: '20220101010102',
+      },
+      {
+        directory: 'migrations',
+        filename: '20220101010103_testing_3.sql',
+        timestamp: '20220101010103',
+      },
+    ];
+    assert.deepEqual(
```

Comment:
The logic for `runAllMigrationsBefore` is plumbed all the way down to `getMigrationsToExecute`.